tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -844,18 +1026,22 @@ def _search_instantiations(
             if not isinstance(node, chapel.FnCall):
                 continue
 
-            rr = node.resolve_via(via) if via else node.resolve()
-            if not rr:
-                continue
-
-            candidate = rr.most_specific_candidate()
-            if not candidate:
-                continue
-
-            sig = candidate.function()
-            fn = sig.ast()
-            if not fn or not isinstance(fn, chapel.Function):
+            resolved = self._resolve_call(node, via)
+            if not resolved:
                 continue
+            fn, sig = resolved
+
+            # Only store the call in the segment table if this is a concrete
+            # functions. There may be multiple instantiations per file,
+            # and until the user has selected one, we shouldn't enable go-to-def
+            # on calls within then.
+            include_in_segments = via is None
+            if include_in_segments:
+                self.call_segments.append(
+                    ResolvedPair(
+                        NodeAndRange(node.called_expression()), NodeAndRange(fn)
+                    )
+                )
```

Comment:
This is the feature I was talking about the other day. Being able to "goto-defintion" on a generic function and have the editor offer the user a list of possible locations is a useful feature. I think you should open an issue for a feature request for this